homeassistant/components/icloud/__init__.py

```diff
@@ -154,7 +155,7 @@ def _get_account(account_identifier: str) -> IcloudAccount:
                     icloud_account = account
 
         if icloud_account is None:
-            raise Exception(
+            raise HomeAssistantError(
```

Comment:
This seems to be a case of bad user input to a service calls. Then we should raise `ValueError`.